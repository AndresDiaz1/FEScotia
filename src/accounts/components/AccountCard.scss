.account-card {
    &__container {
        border: 2px solid black;
        background-color: white;
    }
    &__link {
        text-decoration: underline;
        cursor: pointer;
    }
}