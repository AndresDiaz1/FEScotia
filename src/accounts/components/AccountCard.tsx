import React from 'react';
import { IonGrid, IonLabel, IonRow } from '@ionic/react';
import './AccountCard.scss';
import { Account } from '../AccountTypes';
import { formatCurrency } from '../../utils/Utils';
import { PATHS } from '../../AppConstants';
import { useHistory } from 'react-router';
import { useDispatch } from 'react-redux';
import { fetchDetails } from '../../details/DetailsStore';

const AccountCard: React.FC<Account> = ({ id, accountType, availableValue }) => {
  const history = useHistory();
  const dispatch = useDispatch();

  const goToDetail = () => {
    dispatch(fetchDetails(parseInt(id)));
    history.push(PATHS.DETAILS);
  };

  return (
    <div className="account-card__container ion-padding">
      <IonGrid>
        <IonRow>
          <IonLabel> &middot; {id} </IonLabel>
          <IonLabel> - {accountType}</IonLabel>
        </IonRow>
        <IonRow>
          <IonLabel>{formatCurrency(availableValue)}</IonLabel>
        </IonRow>
        <IonRow>
          <IonLabel>Available amount</IonLabel>
        </IonRow>
        <IonRow className="ion-justify-content-center">
          <IonLabel color="primary" className="account-card__link" onClick={() => goToDetail()}>
            See Details
          </IonLabel>
        </IonRow>
      </IonGrid>
    </div>
  );
};

export default AccountCard;
