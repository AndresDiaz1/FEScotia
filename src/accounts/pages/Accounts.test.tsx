import React from 'react';
import AccountsPage from './Accounts';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import AccountCard from '../components/AccountCard';

describe('Login component', () => {
  describe('Initial state', () => {
    it('Should render No accounts', () => {
      const state = {
        accounts: {
          isLoading: false,
          accounts: [],
          errorMessage: '',
        },
      };
      const store = configureStore([thunk])(state);
      const component = mount(
        <Provider store={store}>
          <AccountsPage />
        </Provider>
      );
      expect(component).toMatchSnapshot();
    });

    it('Should render Error message', () => {
      const state = {
        accounts: {
          isLoading: false,
          accounts: [],
          errorMessage: 'There was a error Testing',
        },
      };
      const store = configureStore([thunk])(state);
      const component = mount(
        <Provider store={store}>
          <AccountsPage />
        </Provider>
      );
      expect(component).toMatchSnapshot();
    });

    it('Should render account cards', () => {
      const state = {
        accounts: {
          isLoading: false,
          accounts: [
            {
              id: '1',
              clientId: 1,
              accountType: 'savings',
              status: 60,
              availableValue: 33,
            },
            {
              id: '2',
              clientId: 1,
              accountType: 'checking',
              status: 65,
              availableValue: 45,
            },
            {
              id: '3',
              clientId: 1,
              accountType: 'credit card',
              status: 17,
              availableValue: 23,
            },
          ],
          errorMessage: '',
        },
      };
      const store = configureStore([thunk])(state);
      const component = mount(
        <Provider store={store}>
          <AccountsPage />
        </Provider>
      );
      const grid = component.find(AccountCard);
      expect(grid.length).toEqual(state.accounts.accounts.length);
    });
  });
});
