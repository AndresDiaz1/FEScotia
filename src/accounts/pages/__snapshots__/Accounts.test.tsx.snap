// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Initial state Should render Error message 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AccountsPage>
    <ForwardRef(IonPage)>
      <IonPage
        forwardedRef={null}
      >
        <div
          className="ion-page"
        >
          <SimpleHeader
            goBack={false}
            title="My Accounts"
          >
            <ForwardRef(IonHeader)>
              <IonHeader
                forwardedRef={null}
              >
                <ion-header>
                  <ForwardRef(IonToolbar)
                    color="scotia-red"
                  >
                    <IonToolbar
                      color="scotia-red"
                      forwardedRef={null}
                    >
                      <ion-toolbar
                        color="scotia-red"
                      >
                        <ForwardRef(IonGrid)>
                          <IonGrid
                            forwardedRef={null}
                          >
                            <ion-grid>
                              <ForwardRef(IonRow)
                                className="ion-justify-content-between"
                              >
                                <IonRow
                                  className="ion-justify-content-between"
                                  forwardedRef={null}
                                >
                                  <ion-row>
                                    <ForwardRef(IonLabel)>
                                      <IonLabel
                                        forwardedRef={null}
                                      >
                                        <ion-label>
                                          My Accounts
                                        </ion-label>
                                      </IonLabel>
                                    </ForwardRef(IonLabel)>
                                    <ForwardRef(IonButton)
                                      className="logout-button"
                                      color="light"
                                      fill="outline"
                                      onClick={[Function]}
                                      shape="round"
                                      size="small"
                                    >
                                      <IonButton
                                        className="logout-button"
                                        color="light"
                                        fill="outline"
                                        forwardedRef={null}
                                        onClick={[Function]}
                                        shape="round"
                                        size="small"
                                      >
                                        <ion-button
                                          color="light"
                                          fill="outline"
                                          onClick={[Function]}
                                          shape="round"
                                          size="small"
                                        >
                                          Logout
                                        </ion-button>
                                      </IonButton>
                                    </ForwardRef(IonButton)>
                                  </ion-row>
                                </IonRow>
                              </ForwardRef(IonRow)>
                            </ion-grid>
                          </IonGrid>
                        </ForwardRef(IonGrid)>
                      </ion-toolbar>
                    </IonToolbar>
                  </ForwardRef(IonToolbar)>
                </ion-header>
              </IonHeader>
            </ForwardRef(IonHeader)>
          </SimpleHeader>
          <ForwardRef(IonContent)>
            <IonContent
              forwardedRef={null}
            >
              <ion-content>
                <ForwardRef(IonLabel)>
                  <IonLabel
                    forwardedRef={null}
                  >
                    <ion-label>
                      You dont have accounts
                    </ion-label>
                  </IonLabel>
                </ForwardRef(IonLabel)>
                <ForwardRef(IonLabel)>
                  <IonLabel
                    forwardedRef={null}
                  >
                    <ion-label>
                      There was a error Testing
                    </ion-label>
                  </IonLabel>
                </ForwardRef(IonLabel)>
                <ForwardRef
                  isOpen={false}
                  message="fetching your accounts..."
                >
                  <IonLoading
                    forwardedRef={null}
                    isOpen={false}
                    message="fetching your accounts..."
                  />
                </ForwardRef>
              </ion-content>
            </IonContent>
          </ForwardRef(IonContent)>
        </div>
      </IonPage>
    </ForwardRef(IonPage)>
  </AccountsPage>
</Provider>
`;

exports[`Login component Initial state Should render No accounts 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AccountsPage>
    <ForwardRef(IonPage)>
      <IonPage
        forwardedRef={null}
      >
        <div
          className="ion-page"
        >
          <SimpleHeader
            goBack={false}
            title="My Accounts"
          >
            <ForwardRef(IonHeader)>
              <IonHeader
                forwardedRef={null}
              >
                <ion-header>
                  <ForwardRef(IonToolbar)
                    color="scotia-red"
                  >
                    <IonToolbar
                      color="scotia-red"
                      forwardedRef={null}
                    >
                      <ion-toolbar
                        color="scotia-red"
                      >
                        <ForwardRef(IonGrid)>
                          <IonGrid
                            forwardedRef={null}
                          >
                            <ion-grid>
                              <ForwardRef(IonRow)
                                className="ion-justify-content-between"
                              >
                                <IonRow
                                  className="ion-justify-content-between"
                                  forwardedRef={null}
                                >
                                  <ion-row>
                                    <ForwardRef(IonLabel)>
                                      <IonLabel
                                        forwardedRef={null}
                                      >
                                        <ion-label>
                                          My Accounts
                                        </ion-label>
                                      </IonLabel>
                                    </ForwardRef(IonLabel)>
                                    <ForwardRef(IonButton)
                                      className="logout-button"
                                      color="light"
                                      fill="outline"
                                      onClick={[Function]}
                                      shape="round"
                                      size="small"
                                    >
                                      <IonButton
                                        className="logout-button"
                                        color="light"
                                        fill="outline"
                                        forwardedRef={null}
                                        onClick={[Function]}
                                        shape="round"
                                        size="small"
                                      >
                                        <ion-button
                                          color="light"
                                          fill="outline"
                                          onClick={[Function]}
                                          shape="round"
                                          size="small"
                                        >
                                          Logout
                                        </ion-button>
                                      </IonButton>
                                    </ForwardRef(IonButton)>
                                  </ion-row>
                                </IonRow>
                              </ForwardRef(IonRow)>
                            </ion-grid>
                          </IonGrid>
                        </ForwardRef(IonGrid)>
                      </ion-toolbar>
                    </IonToolbar>
                  </ForwardRef(IonToolbar)>
                </ion-header>
              </IonHeader>
            </ForwardRef(IonHeader)>
          </SimpleHeader>
          <ForwardRef(IonContent)>
            <IonContent
              forwardedRef={null}
            >
              <ion-content>
                <ForwardRef(IonLabel)>
                  <IonLabel
                    forwardedRef={null}
                  >
                    <ion-label>
                      You dont have accounts
                    </ion-label>
                  </IonLabel>
                </ForwardRef(IonLabel)>
                <ForwardRef
                  isOpen={false}
                  message="fetching your accounts..."
                >
                  <IonLoading
                    forwardedRef={null}
                    isOpen={false}
                    message="fetching your accounts..."
                  />
                </ForwardRef>
              </ion-content>
            </IonContent>
          </ForwardRef(IonContent)>
        </div>
      </IonPage>
    </ForwardRef(IonPage)>
  </AccountsPage>
</Provider>
`;
